lib/rspec/mocks/matchers/receive.rb

```diff
@@ -55,8 +55,9 @@ def setup_any_instance_allowance(subject, &block)
           setup_any_instance_method_substitute(subject, :stub, block)
         end
 
+        own_methods = (instance_methods - superclass.instance_methods)
         MessageExpectation.public_instance_methods(false).each do |method|
-          next if method_defined?(method)
+          next if own_methods.include?(method)
```

Comment:
Wondering if this check is even needed.
The two methods it skipped are `to_s` and `inspe﻿ct` which are defined in `MessageExpectation` and we want to avoid pulling that into `Receive`.
What if we just explicitly state that?
```suggestion
          next if method == :to_s || method == :inspect
```
```

Reply:
Up to you, I don't know rspec-mock much and I rarely use it. 

But yes, that would make sense to me, it would be a much more predictable behavior.

Reply:
The intent is to be future proof.